samcli/lib/config/file_manager.py

```diff
@@ -0,0 +1,99 @@
+"""
+Class to represent the parsing of different file types into Python objects.
+"""
+
+
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+import tomlkit
+
+from samcli.lib.config.exceptions import FileParseException
+
+
+class FileManager(ABC):
+    """
+    Abstract class to be overridden by file managers for specific file extensions.
+    """
+
+    @staticmethod
+    @abstractmethod
+    def read(filepath: Path) -> dict:
+        """
+        Read a file at a given path.
+
+        Parameters
+        ----------
+        filepath: Path
+            The Path object that points to the file to be read.
+
+        Returns
+        -------
+        dict
+            The dictionary representation of the contents at the filepath location.
+        """
+        raise NotImplementedError("Read method not implemented.")
+
+    @staticmethod
+    @abstractmethod
+    def write(document: dict, filepath: Path):
+        """
+        Write a dictionary to a given file.
+
+        Parameters
+        ----------
+        document: dict
+            The object to write.
+        filepath: Path
+            The final location for the document to be written.
+        """
+        raise NotImplementedError("Write method not implemented.")
+
+
+class TomlFileManager(FileManager):
+    """
+    Static class to read and write toml files.
+    """
+
+    @staticmethod
+    def read(filepath: Path) -> dict:
+        """
+        Read a TOML file at the given path.
+
+        Parameters
+        ----------
+        filepath: Path
+            The Path object that points to the file to be read.
+
+        Returns
+        -------
+        dict
+            A Python dictionary representation of the contents of the TOML file at the provided location.
+        """
+        document: dict = {}
+        try:
+            txt = filepath.read_text()
+            document = dict(tomlkit.loads(txt))
+        except OSError:
+            document = {}
```

Comment:
Can we either add a debug log message here when we catch this `OSError` so that we have some trace of this happening when debugging issues?